db/controllers/getTopics.controller.js

```diff
@@ -1,8 +1,25 @@
-const { selectTopics } = require("../models/getTopics.model.js");
-console.log('in the controller');
+const { selectTopics,
+        selectArticleById } = require("../models/getTopics.model.js");
+
 exports.getTopics = (req, res, next) => {
   selectTopics()
     .then((arrayOfTopics) => {
       res.status(200).send({ topics: arrayOfTopics });
     })
 };
+
+exports.getArticleById = (req, res, next) => {
+ const articleId = req.params.article_id;
+ selectArticleById(articleId)
+    .then((articleData) => {
+      if (Object.keys(articleData).length === 0) {
+        res.status(404).send(articleData)
+      }
```

Comment:
This is better handled in your model and sent to your error handling middleware using `Promise.reject`